spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java

```diff
@@ -869,6 +892,10 @@ private boolean rawByParameterIsType(Type parameterType, Type type) {
 		return parameterType instanceof ParameterizedType pType && pType.getRawType().equals(type);
 	}
 
+	public void putInAsyncFailureQueue(BiConsumer<ConsumerRecord<K, V>, RuntimeException> callbackForAsyncFailureQueue) {
```

Comment:
@artembilan 
IMHO, we need to this method. 

The method `setCallbackForAsyncFailure` is in `RecordMessagingMessageListenerAdapter`, which is a concrete implementation of `MessageListener` interface.
However, `MessagingMessageListenerAdapter` is not a concrete class of `MessageListener`.
Therefore, `RecordMessagingMessageListenerAdapter` cannot access fields in its superclass, `MessagingMessageListenerAdapter`.

But yes, as you mentioned, the method name needs to be changed because it gives the wrong impression.

```Java
// RecordMessagingMessageListenerAdapter.java
@Override
public void setCallbackForAsyncFailure(
		BiConsumer<ConsumerRecord<K, V>, RuntimeException> asyncRetryCallback) {
	super.setCallbackForAsyncFailure(asyncRetryCallback);
}

// MessagingMessageListenerAdapter.java
protected void setCallbackForAsyncFailure(BiConsumer<ConsumerRecord<K, V>, RuntimeException> asyncRetryCallback) {
		this.callbackForAsyncFailureQueue = asyncRetryCallback;
}

```
What do you think? Please give me your opinion 🙇‍♂️